### ISSUE TITLE
Error when mutating Error objects in Safari

### ISSUE THREAD

**Issue by justinfagnani** [eslint/espree#282]
Safari does not allow setting of Error.column, as is done is raiseRecorverable:

https://github.com/eslint/espree/blob/88446f3fd0ad21ae09497bbfdd05c0afa06da70c/espree.js#L393

This throws: `Error: Attempted to assign to readonly property.`

**Comment by justinfagnani** [comment]
This seems to be a viable workaround, unless there's some issue with extending from Error that I don't know about. Our test suite passes with it on node, Chrome and Safari.

```javascript
instance.raise = instance.raiseRecoverable = function(pos, message) {
  let loc = getLineInfo(this.input, pos);
  let err = Object.create(new SyntaxError(message), {
    index: {value: pos},
    lineNumber: {value: loc.line},
    column: {value: loc.column + 1},
  });
  throw err;
}
```

**Comment by ilyavolodin** [comment]
What's the use case you are trying to solve? Espree is designed to work in Node, and not Browsers. We only support browsers for online demo of ESLint.

**Comment by justinfagnani** [comment]
We've been running it in browsers for a while in https://github.com/Polymer/hydrolysis

**Comment by ilyavolodin** [comment]
@justinfagnani I'm not saying it wouldn't run, it's just not one of the environments we target specifically. The fix looks reasonable enough for me to add into the codebase, even if it's for unsupported environment, but I don't know enough about Espree to say for sure. @nzakas @kaicataldo @mysticatea do you see any issues with adding this fix in?

**Comment by nzakas** [comment]
We should probably create an `EspreeSyntaxError` constructor that inherits from `SyntaxError` instead. That would be a bit cleaner. 

@justinfagnani can you submit a PR?

**Comment by justinfagnani** [comment]
Yeah, let me look up the CLA situation first.

**Comment by not-an-aardvark** [comment]
Is this still a problem?
